Dockerfile

```diff
@@ -1,34 +1,29 @@
 FROM debian:stretch
 
+RUN groupadd -g 1000 unrealirc && \
+    useradd -r -u 1000 -g unrealirc unrealirc
+
 RUN apt-get update && \
     apt-get upgrade -y && \
     apt-get install -y build-essential git wget libssl-dev && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
-
-RUN git clone https://github.com/unrealircd/unrealircd.git && \
-    cd unrealircd && \
-    git checkout 91e5639a7727c5b55402364a7feeae03e7337e35 && \
-    ./configure \
-      --enable-ssl=/etc/ssl/localcerts/ \
-      --with-showlistmodes \
-      --with-listen=5 \
-      --with-dpath=/etc/unrealircd/ \
-      --with-spath=/usr/bin/unrealircd \
-      --with-nick-history=2000 \
-      --with-sendq=3000000 \
-      --with-bufferpool=18 \
-      --with-permissions=0600 \
-      --with-fd-setsize=1024 \
-      --enable-dynamic-linking && \
+RUN git clone https://github.com/unrealircd/unrealircd.git
```

Comment:
These were all done in one "RUN" block intentionally. If you don't do this docker creates a new "layer" for every operation and you end up with a massive image.